src/profile-logic/profile-data.js

```diff
@@ -1989,46 +2015,145 @@ export function getCategoryPairLabel(
+      if (duration > 0) {
+        newNativeAllocations.time.push(nativeAllocations.time[i]);
+        newNativeAllocations.stack.push(nativeAllocations.stack[i]);
+        newNativeAllocations.duration.push(duration);
+        newNativeAllocations.length++;
+      }
     }
   }
   return newNativeAllocations;
 }
 
 /**
- * See filterToAllocations for detailed documentation. This function filters to only
- * negative values.
+ * This function filters to only negative memory size values in the native allocations.
+ * It shows all of the memory frees.
  */
 export function filterToDeallocations(
   nativeAllocations: NativeAllocationsTable
 ): NativeAllocationsTable {
-  const newNativeAllocations = getEmptyUnbalancedNativeAllocationsTable();
+  let newNativeAllocations;
+  if (nativeAllocations.memoryAddress) {
+    newNativeAllocations = getEmptyBalancedNativeAllocationsTable();
+    for (let i = 0; i < nativeAllocations.length; i++) {
+      const duration = nativeAllocations.duration[i];
+      if (duration < 0) {
+        newNativeAllocations.time.push(nativeAllocations.time[i]);
+        newNativeAllocations.stack.push(nativeAllocations.stack[i]);
+        newNativeAllocations.duration.push(duration);
+        newNativeAllocations.memoryAddress.push(
+          nativeAllocations.memoryAddress[i]
+        );
+        newNativeAllocations.threadId.push(nativeAllocations.threadId[i]);
+        newNativeAllocations.length++;
+      }
+    }
+  } else {
+    newNativeAllocations = getEmptyUnbalancedNativeAllocationsTable();
+    for (let i = 0; i < nativeAllocations.length; i++) {
+      const duration = nativeAllocations.duration[i];
+      if (duration < 0) {
+        newNativeAllocations.time.push(nativeAllocations.time[i]);
+        newNativeAllocations.stack.push(nativeAllocations.stack[i]);
+        newNativeAllocations.duration.push(duration);
+        newNativeAllocations.length++;
+      }
+    }
+  }
+  return newNativeAllocations;
+}
+
+/**
+ * Currently the native allocations naively collect allocations and deallocations.
+ * There is no attempt to match up the sampled allocations with the deallocations.
+ * Because of this, if a calltree were to combine both allocations and deallocations,
+ * then the summary would most likely lie and not misreport leaked or retained memory.
+ * For now, filter to only showing allocations or deallocations.
+ *
+ * This function filters to only positive values.
+ */
+export function filterToRetainedAllocations(
+  nativeAllocations: BalancedNativeAllocationsTable
+): NativeAllocationsTable {
+  // A-----D------A-------D
+  type Address = number;
+  type IndexIntoAllocations = number;
+  const memoryAddressToAllocation: Map<
+    Address,
+    IndexIntoAllocations
+  > = new Map();
+  const retainedAllocation = [];
+  for (
+    let allocationIndex = 0;
+    allocationIndex < nativeAllocations.length;
+    allocationIndex++
+  ) {
+    const duration = nativeAllocations.duration[allocationIndex];
```

Comment:
nit: duration is a really weird name(learned the reason by reading the comment on the type but was pretty confused until that). Can we at least use `byte` or `byteSize` for the variable names to make it more understandable?